pkg/api/pkcs11ca_signing_cert.go

```diff
@@ -35,14 +38,31 @@ func FulcioCASigningCertHandler(subj *challenges.ChallengeResult, publicKey []by
 
 	rootID := []byte(viper.GetString("hsm-caroot-id"))
 
-	// get the existing root CA from the HSM
-	rootCA, err := p11Ctx.FindCertificate(rootID, nil, nil)
-	if err != nil {
-		return "", nil, err
+	// get the existing root CA from the HSM or from disk
```

Comment:
Should we really do this once on process initialization rather than incurring a round trip to the HSM or to the filesystem on a per-transaction level?

Reply:
That's a very valid point, I do wonder if that could be in a follow up patch (as the flawed logic is from the original implementation).